dds/DCPS/Printer.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_PRINTER_HEADER
+#define OPENDDS_DCPS_PRINTER_HEADER
+
+#ifndef OPENDDS_SAFETY_PROFILE
+
+#include "dcps_export.h"
+
+#include <iostream>
+
+#ifndef ACE_LACKS_PRAGMA_ONCE
+#  pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+/**
+ * Class used to help print an IDL Type to a std::ostream. It just configures
+ * the printing options and is what is used directly. The actual work is done
+ * by Printable and operator<< implementations.
+ *
+ * Given an IDL Type T, the following will need to be generated in its
+ * *TypeSupportImpl.cpp by the marshal generator:
+ *   std::ostream& operator<<(const Printable& printable, const Type& value);
+ *
+ * Then given an object of type T called value, it can be printed to stdout
+ * using:
+ *   std::cout << Printer() << value;
+ *
+ * TODO:
```

Comment:
I don't think I'm going to do anymore work on the printing system in this PR since it's not  related to XTypes. It was just useful for the XTypes demo. Past this PR I would like it to develop it a bit for 3.15 more so we can expect it to work on most or all types and get tests and documentation for it.

Reply:
Got it.